src/ragas/metrics/context_relevance.py

```diff
@@ -2,62 +2,137 @@
+    ):
+        self.metric = metric
+        self.cross_encoder = CrossEncoder(model_name)
+
+    def bert_score(self, para1: str, para2: str) -> float:
+        sentences1, sentences2 = sent_tokenize(para1), sent_tokenize(para2)
+        scores = self.cross_encoder.predict(
+            list(product(sentences1, sentences2)), convert_to_numpy=True  # type: ignore
+        )
+        assert isinstance(scores, np.ndarray), "Expects ndarray"
+        scores = scores.reshape(len(sentences1), len(sentences2))
+        return scores.max(axis=1).mean()
+
+    @staticmethod
+    def jaccard_score(para1: str, para2: str) -> float:
+        sentences1, sentences2 = sent_tokenize(para1), sent_tokenize(para2)
+        intersect = len(np.intersect1d(sentences1, sentences2))
+        union = len(np.union1d(sentences1, sentences2))
+        return intersect / union
+
+    def evaluate(self, answers: List[List[str]]) -> float:
+        """
+        eval nC2 combinations
+        """
+        scores = []
+        groups = combinations(answers, 2)
+        for group in groups:
+            if self.metric == "jaccard":
+                score = self.jaccard_score(*group)  # type: ignore
+            elif self.metric == "bert_score":
+                score = self.bert_score(*group)  # type: ignore
+            else:
+                score = 0
+                raise ValueError(f"Metric {self.metric} unavailable")
+            scores.append(score)
+        scores = np.mean(scores)
+        assert isinstance(scores, float), "Expects floats"
+        return scores
 
 
 @dataclass
 class ContextRelevancy(Metric):
-    batch_size: int = 32
     name: str = "context_relavency"
-    model_name: str = "t5-base"
+    batch_size: int = 15
+    agreement_metric: str = "bert_score"
+    strictness: int = 2
+    model_name: str = "cross-encoder/stsb-TinyBERT-L-4"
 
     def init_model(self: t.Self):
-        self.device = "cuda" if torch.cuda.is_available() else "cpu"
-        self.model = QGen(self.model_name, self.device)
-
-    @staticmethod
-    def _make_question_context_rank_pairs(row: dict) -> dict:
-        q_id = row["q_id"]
-        q = row["question"]
-        cs = row["contexts"]
-        new_q_ids = []
-        sentences = []
-        for i, q_id in enumerate(q_id):
-            for c in cs[i]:
-                sentences.append([q[i], c])
-                new_q_ids.append(q_id)
-
-        return {"q_id": new_q_ids, "sentences": sentences}
+        self.sent_agreement = SentenceAgreement(
+            model_name=self.model_name, metric=self.agreement_metric
+        )
 
     def score(self: t.Self, dataset: Dataset) -> Dataset:
         """
         dataset: Dataset[question: list[str], contexts: list[list[str]]]
         """
+        prompts = []
+        questions, contexts = dataset["question"], dataset["contexts"]
+        for q, c in zip(questions, contexts):
+            prompt = CONTEXT_RELEVANCE.format(q, "\n".join(c))
+            prompts.append(prompt)
```

Comment:
wouldn't this need batching. If there are more than 20 (questions, contexts) pairs openAI will return an error?